### ISSUE TITLE
[Bug] Autoplay blocking blocks player on lfm.ch, even if initiated by the user

### ISSUE THREAD

**Issue by denschub** [mozilla-mobile/fenix#16402]
## Steps to reproduce

1. Open https://www.lfm.ch/player/popup/index.html
2. Click the play icon

### Expected behavior

Stream starts.

### Actual behavior

Stream does not start. Instead, you'll see a autoplay blocking message in the console:

> Autoplay is only allowed when approved by the user, the site is activated by the user, or media is muted.

Even though the user just clicked on the play button. The site is probably doing some fancy JS magic to play the video and we don't realize the play was started by a DOM event, but maybe this is fixable.

### Device information

* Android device: (emulator)
* Fenix version: Nightly 201104 17:01 (Build #2015773709)

**Comment by Amejia481** [comment]
@denschub we did some improvements related to autoplay could you verify if this is still an issue in [nightly](https://play.google.com/store/apps/details?id=org.mozilla.fenix)? I will also as to the QA team to verify.

**Comment by abodea** [comment]
Verified as fixed on the latest **Nightly 18/11** with **Google Pixel 4 XL (11).**
Note that the stream started without problems!

**Comment by denschub** [comment]
Mh, it's still broken in Nightly 201117 20:41 (Build #2015776237), and Google Play does not offer me the new nightly yet, but I trust you :)

**Comment by abodea** [comment]
Hi, @denschub can you try with the build from [here?]([url](https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.nightly.latest/armeabi-v7a))

**Comment by denschub** [comment]
Using `Nightly 201118 05:01 (Build #2015776301)`, and a freshly reset profile: Same behavior, autoplay blocking exception still in the console:

<img width="1591" alt="Screenshot 2020-11-18 at 18 12 47" src="https://user-images.githubusercontent.com/344777/99563725-d72abe00-29c9-11eb-8c30-1b74966cd55e.png">

So this probably still is there, although I'm not sure why it works in your tests, @abodea. Did you toggle the autoplay setting to a non-default value or something?

**Comment by Amejia481** [comment]
@denschub did you switch preferences to allow the content to be played?
![image](https://user-images.githubusercontent.com/773158/99571093-ddf10b00-29a0-11eb-92a4-c6a1c8e5403c.png)

**Comment by denschub** [comment]
@Amejia481 I did not - and I should not have to, which kinda is the point of the issue? :) The player in question is not attempting to auto-play anything, it wants to start playing after the user clicked on a play button.

The autoplay detection shouldn't even be in effect here, but the site is probably doing some JS trickery (maybe manually dispatching another click event to somewhere?) that makes Fenix *think* the site wants to autoplay something, while the site is actually just handling user interaction.

**Comment by Amejia481** [comment]
Got it, let's reopen and take a second look, thanks :)

**Comment by denschub** [comment]
\*gently pokes this issue\*

**Comment by denschub** [comment]
The site has switched designs, but this still reproduces. Same STR.

**Comment by Amejia481** [comment]
👋🏽 
After testing couple of times I noticed the behavior is not reliable sometimes it works and sometimes it doesn't. I noticed from Fenix/AC we are getting and returning the correct value for the auto play permissions, we don't have control over the heuristics to unblock the content when the users clicks on the play icon, I filed bugzilla ticket for the Audio/Video: Playback           to take a look https://bugzilla.mozilla.org/show_bug.cgi?id=1742465.

I'm closing this ticket for now, as there is not much that we can do on our end.
